### DISCUSSION TITLE
gobind failed

### DISCUSSION THREAD

**Discussion by shahinfasihi** [v2fly/v2ray-core#2378] [Q&A]
I am trying to generate a `.aar` file with the latest code of this repository but I get an error that I will mention below
go version => 1.20
ndk version => 25.2.9519653

`sudo gomobile bind -v -androidapi 19 -ldflags='-s -w' ./`
`sudo gomobile bind -target=android -v -o v2ray.aar ./`

# gobind/gobind
core_android.c:272:1: error: redefinition of 'Java_core_InboundHandlerConfig_getTag'
core_android.c:235:1: note: previous definition is here
core_android.c:377:1: error: redefinition of 'Java_core_OutboundHandlerConfig_getTag'
core_android.c:340:1: note: previous definition is here
core_android.c:396:1: error: redefinition of 'Java_core_OutboundHandlerConfig_getExpire'
core_android.c:328:1: note: previous definition is here
core_android.c:411:1: error: redefinition of 'Java_core_OutboundHandlerConfig_getComment'
core_android.c:320:1: note: previous definition is here
./gomobile: go build -v -buildmode=c-shared -o=/tmp/gomobile-work-3767676161/android/src/main/jniLibs/x86/libgojni.so ./gobind failed: exit status 1

**Comment by rurirei** [comment]
>> core_android.c:272:1: error: redefinition of 'Java_core_InboundHandlerConfig_getTag'
core_android.c:235:1: note: previous definition is here
core_android.c:377:1: error: redefinition of 'Java_core_OutboundHandlerConfig_getTag'

you should implement the interfaces you own correctly, as gomobile accepts only the interfaces within the top-level package.

**Reply by shahinfasihi** [reply]
thank you for your reply 
but I didn't get you correctly; I am not trying to do the code part yet I need a jni base .so file to import into my android studio project 
but I get an error when trying to generate the `.aar` file 
the error in the `core_android.c ` file is indicating the error when trying to bind Go objects to Java

**Reply by rurirei** [reply]
you cannot simply use this into the android project.

**Reply by shahinfasihi** [reply]
i couldn't find the correct documentation could you please help me ?

**Reply by mydogshitgold** [reply]
If you are trying to build a V2Ray front for Android you can use [2dust/v2rayNG](https://github.com/2dust/v2rayNG) and [2dust/AndroidLibV2rayLite](https://github.com/2dust/AndroidLibV2rayLite) as examples.

**Reply by shahinfasihi** [reply]
I did not get my answer I checked the AndroidLibV2rayLite before but I want to get the .aar from the main branch of this repo is it possible or not ? I just want to start a new app so I want to check all the possible ways and then start to code

**Reply by mydogshitgold** [reply]
> I checked the AndroidLibV2rayLite before but I want to get the .aar from the main branch of this repo, is it possible or not?

I found your question really strange. Have you **read** the actual code or have you just been lurking around? [2dust/AndroidLibV2rayLite](https://github.com/2dust/AndroidLibV2rayLite) is a wrapper. You can also write one of these for yourself.

**Reply by shahinfasihi** [reply]
I know that but thanks for your reply
if you check the first question that I asked I think you won't find it strange I am asking about the error that happened with this repo 
so correct me if I am wrong, generating a .aar file from the v2fly repo is a mistake? is it a wrong idea?

**Reply by mydogshitgold** [reply]
> so correct me if I am wrong, generating a .aar file from the v2fly repo is a mistake? is it a wrong idea?

What could possibly make you think that generate binding to **176** structures, **50** interfaces, **1050** functions, **3230** methods and **28959** lines of generated protobuf-go file with conflicting symbols would be a good idea?

And why would developers of [2dust/v2rayNG](https://github.com/2dust/v2rayNG) use wrappers instead directly bind to v2ray?
